src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -569,187 +418,207 @@ try
 CATCH_RETURN()
 
 // Routine Description:
-// - Attempts to locate the font given, but then begins falling back if we cannot find it.
-// - We'll try to fall back to Consolas with the given weight/stretch/style first,
-//   then try Consolas again with normal weight/stretch/style,
-//   and if nothing works, then we'll throw an error.
+// - Build the needed data for rendering according to the font used
 // Arguments:
-// - familyName - The font name we should be looking for
-// - weight - The weight (bold, light, etc.)
-// - stretch - The stretch of the font is the spacing between each letter
-// - style - Normal, italic, etc.
+// - desired - Information specifying the font that is requested
+// - actual - Filled with the nearest font actually chosen for drawing
+// - dpi - The DPI of the screen
 // Return Value:
-// - Smart pointer holding interface reference for queryable font data.
-[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFace1> DxFontRenderData::_ResolveFontFaceWithFallback(std::wstring& familyName,
-                                                                                                      DWRITE_FONT_WEIGHT& weight,
-                                                                                                      DWRITE_FONT_STRETCH& stretch,
-                                                                                                      DWRITE_FONT_STYLE& style,
-                                                                                                      std::wstring& localeName) const
-{
-    auto face = _FindFontFace(familyName, weight, stretch, style, localeName);
+// - None
+void DxFontRenderData::_BuildFontRenderData(const FontInfoDesired& desired, FontInfo& actual, const int dpi)

```

Comment:
Might use the split mode to review this file from this point onwards.

Reply:
This method sets `_fontSize`, `_lineSpacing`, `_lineMetrics` and `_glyphCell ` internally. I wish all of them can be explicitly acquired just like the  `DefaultTextFormat()` or `Analyzer()`, but I failed to find a way to break this method into smaller ones, because the logic inside this is tightly-coupled and I honestly understand none of it.

Reply:
Then it's done! I can live with all the logic wrapped inside ` _BuildFontRenderData(desired, actual, dpi)` . The caller does not need to care about the details inside, and can enjoy the much clearer interface.